CRM/Contribute/Form/CancelSubscription.php

```diff
@@ -203,8 +203,18 @@ public function postProcess() {
     }
 
     if (CRM_Utils_Array::value('send_cancel_request', $params) == 1) {
-      $cancelParams = ['subscriptionId' => $this->_subscriptionDetails->subscription_id];
-      $cancelSubscription = $this->_paymentProcessorObj->cancelSubscription($message, $cancelParams);
+      // Note the 'subscriptionId' here is the value stored in
+      // civicrm_contribution_recur.processor_id
+      $cancelParams = [
+        'subscriptionId'      => $this->_subscriptionDetails->subscription_id,
+        'contributionRecurID' => $this->_subscriptionDetails->recur_id,
```

Comment:
@artfulrobot Whilst I approve of adding this we should do it as a separate PR please!